### DISCUSSION TITLE
Improving the support of LaTeX equations in Typst output

### DISCUSSION THREAD

**Discussion by oliviermeslin** [quarto-dev/quarto-cli#11075] [Feature Requests]
### Description

__TL;DR: I wonder whether it would be possible to include the `mitex` Typst package in Quarto (or in Pandoc) to make it easier to produce a website and a Typst pdf from the same Quarto files.__

-----

## Context
I'm working on a Quarto project, and I'd like to produce two outputs: a website and a pdf produced by Typst. There are many equations in this project, written using the LaTeX syntax. I can easily include them in the Typst document using the [`mitex`](https://typst.app/universe/package/mitex/) package, but then the qmd files cannot be used directly to produce the website. Some workarounds using `when-format` might be possible, but they would be detrimental to the readibility of the qmd files and would likely result in code duplication.

Here is a MWE of my current source code for the Typst output:

````
---
format: typst
---

```{=typst}
#import "@preview/mitex:0.2.4": *
```

This a Typst document with a simple inline LaTeX equation (`#mi("$\hat{F}$")`{=typst}) and a complicated LaTeX block equation:

```{=typst}
#mitex(`
$\begin{align*}
\mathcal{\tilde{L}}^{(k)}
&= \sum_{j=1}^{T} \sum_{i\in I_j} \Bigl[g_i f_k(\mathbf{x}_i) &+ \frac{1}{2} h_i f^2_k(\mathbf{x}_i)\Bigr] &+ \gamma T + \frac{1}{2} \lambda \sum_{j=1}^T w_j^2 \\
&= \sum_{j=1}^{T} \sum_{i\in I_j} \Bigl[g_i w_j &+ \frac{1}{2} h_i w_j^2\Bigl] &+ \gamma T + \frac{1}{2} \lambda \sum_{j=1}^T w_j^2 \\
&= \sum^T_{j=1} \Bigl[w_j\Bigl(\sum_{i\in I_j} g_i\Bigr) &+ \frac{1}{2} w_j^2 \Bigl(\sum_{i\in I_j} h_i + \lambda\Bigr) \Bigr] &+ \gamma T  \label{eq:dudu}
\end{align*}$
  `)
```

````

## Feature request

I would like to produce a Typst document with LaTeX-style equations without having to encapsulate them explicitly in the proper `#mitex()` or `#mi()` environments in the qmd files. This way, the same qmd files could be used to produce the pdf and the website.

Based on this [discussion](https://github.com/quarto-dev/quarto-cli/discussions/9406), I think that a solution could consist in adding a Lua filter in the Quarto Typst support (or in Pandoc), so that LaTeX-style equations would invisibly be encapsulated using `mitex` when output format is Typst. Maybe this feature request concerns Pandoc more than Quarto, let me know if you think so.

**Comment by mcanouil** [comment]
You should be able to do that with a LUA filter targeting equation in the AST.

Note that adding a dependency is not something to be done lightly especially from a project that is very young (Typst).

Edit: see @cderv's comment with an experimental extension showing the above idea: https://github.com/quarto-dev/quarto-cli/discussions/11075#discussioncomment-10959511

**Comment by cderv** [comment]
I wanted to see how easy it would be to integrate with mitex using an extension. Here is an experiment you can use and contribute to improve
- https://github.com/quarto-ext/mitex

**Reply by oliviermeslin** [reply]
Thanks @cderv ! I'll definitely have a look at this extension! 

By the way, I had already started writing a [Lua filter](https://github.com/oliviermeslin/DT_methodes_ensemblistes/blob/main/lua/typstify_equations.lua) that encapsulates equations in the proper mitex environments.
